src/hooks/api.tsx

```diff
@@ -91,22 +117,115 @@ export function useGovernorCouncil(
 export function useProposals(
   governorAddress: string | undefined,
   currentBlock: number | undefined,
-  enabled: boolean = true,
+  enabled: boolean = true
 ): UseQueryResult<Proposal[]> {
   const { network } = useWallet();
-  const paramsDefined = governorAddress !== undefined && currentBlock !== undefined;
+  const paramsDefined =
+    governorAddress !== undefined && currentBlock !== undefined;
 
   async function loadProposals(): Promise<Proposal[]> {
     if (paramsDefined) {
-      const proposals = await fetchProposalsByGovernor(governorAddress);
+      const proposals = (
+        await fetchProposalsByGovernor(governorAddress)
+      ).reverse();
       const updatedProposals: Proposal[] = [];
+      let proposalsToCheck = 5;
+      let nextProposalId = (await getNextPropId(network, governorAddress)) ?? 0;
+      let id = proposals[0] ? proposals[0].id : nextProposalId - 1;
+
+      // fetch proposals from ledger if not available in indexer
+      for (let i = id; i < nextProposalId - 1; i++) {
+        let voteCount = await getProposalVotes(network, governorAddress, i);
+        let proposalFromLedger = await getProposal(network, governorAddress, i);
+        if (proposalFromLedger) {
+          updatedProposals.push({
+            id: proposalFromLedger.id,
+            status: proposalFromLedger.data.status as number,
+            title: proposalFromLedger.config.title,
+            description: proposalFromLedger.config.description,
+            action: proposalFromLedger.config.action,
+            proposer: proposalFromLedger.data.creator,
+            vote_start: proposalFromLedger.data.vote_start,
+            vote_end: proposalFromLedger.data.vote_end,
+            eta: proposalFromLedger.data.eta,
+            vote_count: voteCount,
+          });
+        }
```

Comment:
Is `getProposal` used elsewhere? It seems like it should do all of this at once, and return our normalized proposal object used throughout the frontend.

We likely would be better off performing both promises at the same time too, instead of doing them sequentially.